lib/msf/core/exe/segment_appender.rb

```diff
@@ -16,6 +16,12 @@ def payload_stub(prefix)
     def generate_pe
       # Copy our Template into a new PE
       pe_orig = Metasm::PE.decode_file(template)
+      if is_warbird?(pe_orig)
```

Comment:
In my understanding, `warbird` (license verification) only applies to `SegmentInjector` (see this [PR](https://github.com/rapid7/metasploit-framework/pull/5660) and [issue](https://github.com/rapid7/metasploit-framework/issues/4380)).
Did you experienced issues with `SegmentAppender` and binaries with `warbird`?

Reply:
So my thought here was that `SegementAppender` is called if the `-k` flag is also supplied. If just `-x` is used then `SegmentInjector` is used. Since the two could technically be called on the same binary my thought was that similar checks should be performed. I also thought this would be logical given that the code seems to be fairly similar for both.

TLDR: This change was more based off of some assumptions from observations I found whilst doing this PR and I haven't tested it fully. If you feel this doesn't belong here then feel free to remove it.

Reply:
I haven't tested it but, in my understanding, it looks like it is the way around. `SegmentInjector` is used when the `-k` flag is supplied. Without this flag, `SegmentAppender` is used.

The `generate_pe` method has a different behavior in each class:
- `SegmentAppender::generate_pe` will add the payload to a new section and add it to the binary. It also updates the entry point to jump to this new section.
- `SegmentInjector::generate_pe` will replace the original `.text` section with the payload and do some magic to execute it. This is probably why `warbird` protection is not compatible with this kind of injection.

Reply:
Makes sense, happy to undo this edit based on what you have said.

Reply:
Will be undone with next commit, sorry about that!